### PR TITLE
 feat: enable Ubuntu 24.04 as a development base

### PR DESCRIPTION
**Pull request by lengau**
Requires #1554 

Only the last commit is actually this PR

### CODE REVIEW

File: charmcraft/const.py [hunk 1]
```diff
@@ -53,12 +55,26 @@
...
+    "ubuntu@20.04",
+    "ubuntu@22.04",
+    "ubuntu@23.10",
+    "centos@7",
+    "almalinux@9",
+]
+BaseStr = CommonBaseStr | Literal["ubuntu@24.04"]
+BuildBaseStr = CommonBaseStr | Literal["ubuntu@devel"]
+
+DEVEL_BASE_STRINGS = ("ubuntu@24.04",)  # Bases that require a specified build base.
```
Comment (sergiusens) [inline charmcraft/const.py:69]: the point was that from today we would not require build-base for 24.04

**Review by tigarmo — APPROVED** [review]

**Review by sergiusens — CHANGES_REQUESTED** [review]
Nice work, while I did not review the entirety of this, I did spot that the build-base logic of devel for 24.04 is not a requirement starting today

File: charmcraft/const.py (hunk 1 above)
Comment (sergiusens) [reply charmcraft/const.py:69]: new information makes my comment no longer valid

Comment (lengau) [reply charmcraft/const.py]: Ohh right! Fixed :-)

**Review by syu-w — APPROVED** [review]

**Pull request merged**